### PR TITLE
Refactor Client class, write to file, and property based testing

### PR DESCRIPTION
**Pull request by nbuonin**
This commit primarily adds the functionality to write out the clients to
a CSV file. This functionality is now the default. If a user wishes to
have Mailchimp send a file, you'll have to set it as a config option.

This commit also refactors the Client class constructor to reduce the
coupling bewteen Client and the file row object. It now takes in the
params one value at a time, rather than reading a whole row. The row
could be malformed, and it doesn't make much sense to have the class
validate this.

Lastly, this commit introduces property based testing for the new write
to file functionality. It uses a combination of Hypothesis to generate
randomized Client objects and mocking to capture calls to the OS.

### CODE REVIEW

File: mailchimp_subscriber.py [hunk 1]
```diff
@@ -56,9 +60,18 @@ def load_conf(conf_file):
     containing the MC List Id, MC User, and MC API key"""
     config = configparser.ConfigParser()
     config.read(conf_file)
+    send_mc_email = False
+    try:
+        if (config['DEFAULT'].getboolean('SendMCEmail',
+                                         fallback=False)):
+            send_mc_email = True
+    except ValueError:
+        pass
```
Comment (sdreher) [inline mailchimp_subscriber.py]: I see you're using a getter with a default value specified. Does this actually throw an exception?

File: mailchimp_subscriber.py [hunk 2]
```diff
@@ -70,7 +83,10 @@ def load_users(users_file):
             # Note that this isn't testing for multiple appearances of the same
             # client. If theres more than one, it takes the last appearence.
             try:
-                clients[row[EMAIL_COL]] = Client(row)
+                if (len(row) >= 3):
+                    clients[row[EMAIL_COL]] = Client(row[EMAIL_COL],
```
Comment (sdreher) [inline mailchimp_subscriber.py]: nit: this is a bit confusing to read. Suggestion on a clearer, but more verbose approach:
```
client = Client(...)
clients[client.email_address] = client

File: mailchimp_subscriber.py [hunk 3]
```diff
@@ -81,23 +97,43 @@ def process_users(users, list_id, mc_user, mc_key):
     """Takes a list of emails and returns a list of those not subscribed
     to the MailChimp list"""
     mc_client = MailChimp(mc_user, mc_key)
-    for client in users.values:
+    for client in users.values():
         set_mailchimp_status(client, mc_client, list_id)
-        add_user(client, mc_client, list_id)
+
+    if (SEND_MC_EMAIL):
```
Comment (sdreher) [inline mailchimp_subscriber.py]: At line 65, the config.default is used to read this configuration option. Should that be used consistently here?

File: mailchimp_subscriber.py [hunk 4]
```diff
@@ -81,23 +97,43 @@ def process_users(users, list_id, mc_user, mc_key):
...
+    for client in clients:
+        if (client.mailchimp_status == 'pending' or
+                client.mailchimp_status == 'not_present'):
+            try:
+                mc_client.update.list.members.update(list_id,
+                                                     client.email_hash,
+                                                     "{'status': 'pending'}")
+                return True
+            except requests.exceptions.HTTPError:
+                return False
```
Comment (sdreher) [inline mailchimp_subscriber.py]: Should this fail for the whole list if just one fails?

File: mailchimp_subscriber.py (hunk 3 above)
Comment (nbuonin) [reply mailchimp_subscriber.py]: replace with config dict

**Comment by sdreher** [comment]
This PR just has a few flake8 errors to resolve.

File: mailchimp_subscriber.py (hunk 3 above)
Comment (sdreher) [reply mailchimp_subscriber.py]: Does this still need to be replaced?

File: mailchimp_subscriber.py (hunk 1 above)
Comment (sdreher) [reply mailchimp_subscriber.py]: also...does this throw an exception?

Comment (nbuonin) [reply mailchimp_subscriber.py]: Nope, it doesn't, I'll get rid of the try/except

**Pull request merged**